sql/migrate/migrate.go

```diff
@@ -171,6 +190,40 @@ type (
 
 	// PlannerOption allows managing a Planner using functional arguments.
 	PlannerOption func(*Planner)
+
+	// A Revision denotes an applied migration in a deployment. Used to track migration executions state of a database.
+	Revision struct {
+		// Version of the migration.
+		Version string
+		// Description of this migration.
+		Description string
+		// ExecutionState of this migration. One of ["ongoing", "ok", "error"].
+		ExecutionState string
+		// ExecutedAt denotes when this migration was started to be executed.
+		ExecutedAt time.Time
+		// ExecutionTime denotes the time it took for this migration to be applied on the database.
+		ExecutionTime time.Duration
+		// Error holds information about a migration error (if occurred).
+		Error string
```

Comment:
Can you document here the error formats you use in the code (e.g. "Go", "Statement").